packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableRowAction.tsx

```diff
@@ -81,19 +97,53 @@ export function TransactionsTableRowAction({
     }
   }
 
-  if (tx.status === 'Unconfirmed') {
+  if (isDepositReadyToRedeem(tx)) {
+    // Failed retryable
     return (
       <Tooltip
+        show={isRedeemButtonDisabled}
         wrapperClassName=""
-        content={<span>Funds aren&apos;t ready to claim yet</span>}
+        content={
+          <span>
+            Please connect to the L2 network to re-execute your deposit. You
+            have 7 days to re-execute a failed tx. After that, the tx is no
+            longer recoverable.
+          </span>
+        }
       >
-        <Button variant="primary" disabled>
-          Claim
+        <Button
+          variant="primary"
+          loading={isRedeeming}
+          disabled={isRedeemButtonDisabled}
+          onClick={() => redeem(tx)}
+          className="bg-red-400 p-2 text-xs"
+        >
+          Retry
         </Button>
       </Tooltip>
     )
   }
 
+  if (
+    tx.status === 'Unconfirmed' ||
+    tx.depositStatus === DepositStatus.L1_PENDING ||
+    tx.depositStatus === DepositStatus.L2_PENDING ||
+    typeof cctpRemainingTime !== 'undefined'
+  ) {
+    return (
+      <div className="flex flex-col text-center text-[10px]">
```

Comment:
i know that 10px is what the design specified but it's way too small for users and is bad a11y practice

can we talk to designer about this?

Reply:
You're right. I changed to `text-xs`.

I will aggregate all those issues to forward to the designer, but for now we can implement those small changes independently so we're not blocked.